instrumentation/gruf/lib/opentelemetry/instrumentation/gruf/interceptors/client.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+module OpenTelemetry
+  module Instrumentation
+    module Gruf
+      module Interceptors
+        class Client < ::Gruf::Interceptors::ClientInterceptor
+          # rubocop:disable Metrics/MethodLength
+          def call(request_context:)
+            return yield if instrumentation_config.blank?
+
+            service = request_context.method.split("/")[1]
+            method = request_context.method_name
+            method_name_with_service = [service.underscore, method].join(".").downcase
+
+            if instrumentation_config[:grpc_ignore_methods_on_client].include?(method_name_with_service)
+              return yield
+            end
+
+            metadata = request_context.metadata
+            attributes = {
+              OpenTelemetry::SemanticConventions::Trace::RPC_SYSTEM => 'grpc',
+              OpenTelemetry::SemanticConventions::Trace::RPC_SERVICE => service,
+              OpenTelemetry::SemanticConventions::Trace::RPC_METHOD => method,
+              OpenTelemetry::SemanticConventions::Trace::PEER_SERVICE => instrumentation_config[:peer_service],
+              'rpc.type' => request_context.type.to_s,
+              'net.sock.peer.addr' => request_context.call.instance_variable_get(:@wrapped)&.peer
+            }.compact
+
+            attributes.merge!(allowed_metadata_headers(metadata.stringify_keys))
+
+            instrumentation_tracer.in_span(request_context.method.to_s,
+              attributes: attributes,
+              kind: OpenTelemetry::Trace::SpanKind::CLIENT,
+            ) do |span|
+              OpenTelemetry.propagation.inject(metadata)
+              yield.tap do
+                span&.set_attribute(OpenTelemetry::SemanticConventions::Trace::RPC_GRPC_STATUS_CODE, 0)
+              end
+            rescue Exception => e
```

Comment:
Did rubocop miss this? 

```suggestion
            rescue StandardError => e
```

I'd hesitate here rescue Exception because it will inadvertently catch fatal things like NoMemoryErrors. 

In cases like that the Ruby process will likely be OOM killed and the traces won't make it out of the app anyway

Reply:
Ok, I fixed to StandartError